basculehttp/bearerTokenFactory.go

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright 2021 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package basculehttp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/xmidt-org/arrange"
+	"github.com/xmidt-org/bascule"
+	"github.com/xmidt-org/bascule/key"
+	"go.uber.org/fx"
+)
+
+const (
+	jwtPrincipalKey = "sub"
+)
+
+var (
+	ErrorInvalidPrincipal = errors.New("invalid principal")
+	ErrorInvalidToken     = errors.New("token isn't valid")
+	ErrorUnexpectedClaims = errors.New("claims wasn't MapClaims as expected")
+
+	ErrNilResolver = errors.New("resolver cannot be nil")
+)
+
+// BearerTokenFactory parses and does basic validation for a JWT token,
+// converting it into a bascule Token.
+type BearerTokenFactory struct {
```

Comment:
Since this is the actual object that implements the `TokenFactory` interface, I wasn't going to...but I'm open to it! I'm not sure what's normal under those circumstances.

Reply:
I see... Either way works for me. There's also the option to split this into a parameter struct `BearerTokenFactoryIn` and the implementation struct `BearerTokenFactory` which instead of a `key.Resolver` has the built out `keyFunc` 😃 
It could make ParseAndValidate slightly shorter 🕺

Reply:
I tried pulling out the `keyFunc` but found an issue with `ParseAndValidate()` using its context parameter in the key function.  Because of this, I'm going to leave it the way it is now.